### PR TITLE
feat: add data export button in settings page

### PR DESCRIPTION
**Pull request by mariembencheikh**
#318 

[exportData.webm](https://github.com/user-attachments/assets/85842fba-fd63-4308-ba63-81fcbf70864f)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "@emotion/cache": "11.13.0",
     "@emotion/react": "11.13.0",
     "@emotion/styled": "11.13.0",
-    "@graasp/query-client": "3.20.3",
+    "@graasp/query-client": "https://github.com/graasp/graasp-query-client#874-add-member-export-data",
```
Comment (ReidyT) [inline package.json:17]: Just a reminder ^^

File: src/config/notifier.ts [hunk 2]
```diff
@@ -57,6 +58,10 @@ export default ({
       message = getErrorMessageFromPayload(payload);
       break;
     }
+    case exportMemberDataRoutine.FAILURE: {
+      message = i18n.t(payload?.message ?? 'EXPORT_ERROR_MESSAGE');
```
Comment (ReidyT) [inline src/config/notifier.ts:62]: Maybe you can use a const here for `EXPORT_ERROR_MESSAGE` ?

File: src/config/notifier.ts [hunk 3]
```diff
@@ -71,6 +76,10 @@ export default ({
       );
       break;
     }
+    case exportMemberDataRoutine.SUCCESS: {
+      message = i18n.t(payload?.message ?? 'EXPORT_SUCCESS_MESSAGE');
```
Comment (ReidyT) [inline src/config/notifier.ts:80]: use a const here too ?

File: src/langs/en.json [hunk 4]
```diff
@@ -106,5 +106,10 @@
   "MEMBER_STORAGE_PARENT_FOLDER": "Parent folder",
   "NO_PARENT": "-",
   "STORAGE_FILES_ERROR": "Oops, an error occurred while fetching storage files. Please try again later.",
-  "STORAGE_FILES_EMPTY": "You didn't upload any file."
+  "STORAGE_FILES_EMPTY": "You didn't upload any file.",
+  "EXPORT_INFORMATIONS_TITLE": "Export data",
+  "EXPORT_INFORMATIONS_DESCRIPTION": "You can download a copy of your data",
+  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Start exporting",
+  "EXPORT_SUCCESS_MESSAGE": "Your request has been sent. An email will be sent to you in several minutes.",
```
Comment (ReidyT) [inline src/langs/en.json:113]: I'd write something like the following (it is only a suggestion :) ).

```suggestion
  "EXPORT_SUCCESS_MESSAGE": "You have successfully requested your data. You will receive an e-mail in a few minutes.",
```

File: src/langs/fr.json [hunk 5]
```diff
@@ -106,5 +106,10 @@
   "MEMBER_STORAGE_PARENT_FOLDER": "Dossier parent",
   "NO_PARENT": "-",
   "STORAGE_FILES_ERROR": "Oops, une erreur est survenue lors de la récupération de la liste des fichiers. Veuillez réessayer plus tard.",
-  "STORAGE_FILES_EMPTY": "Vous n'avez pas téléchargé de fichier."
+  "STORAGE_FILES_EMPTY": "Vous n'avez pas téléchargé de fichier.",
+  "EXPORT_INFORMATIONS_TITLE": "Export des données",
+  "EXPORT_INFORMATIONS_DESCRIPTION": "Vous pouvez télécharger une copie de vos données",
+  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Commencez à exporter",
+  "EXPORT_SUCCESS_MESSAGE": "Votre demande a été envoyée. Vous recevrez un email dans quelques minutes.",
```
Comment (ReidyT) [inline src/langs/fr.json:113]:
```suggestion
  "EXPORT_SUCCESS_MESSAGE": "Vous avez demandé vos données avec succès. Vous recevrez un e-mail dans quelques minutes.",
```

File: src/langs/en.json [hunk 6]
```diff
@@ -106,5 +106,10 @@
   "MEMBER_STORAGE_PARENT_FOLDER": "Parent folder",
   "NO_PARENT": "-",
   "STORAGE_FILES_ERROR": "Oops, an error occurred while fetching storage files. Please try again later.",
-  "STORAGE_FILES_EMPTY": "You didn't upload any file."
+  "STORAGE_FILES_EMPTY": "You didn't upload any file.",
+  "EXPORT_INFORMATIONS_TITLE": "Export data",
+  "EXPORT_INFORMATIONS_DESCRIPTION": "You can download a copy of your data",
+  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Start exporting",
```
Comment (ReidyT) [inline src/langs/en.json:112]: I'm wondering if `Start exporting` is the best choice here. I'd write something like the following. I don't know what do you think and what the other thinks ?

```suggestion
  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Request your data",
```

File: src/langs/fr.json [hunk 7]
```diff
@@ -106,5 +106,10 @@
   "MEMBER_STORAGE_PARENT_FOLDER": "Dossier parent",
   "NO_PARENT": "-",
   "STORAGE_FILES_ERROR": "Oops, une erreur est survenue lors de la récupération de la liste des fichiers. Veuillez réessayer plus tard.",
-  "STORAGE_FILES_EMPTY": "Vous n'avez pas téléchargé de fichier."
+  "STORAGE_FILES_EMPTY": "Vous n'avez pas téléchargé de fichier.",
+  "EXPORT_INFORMATIONS_TITLE": "Export des données",
+  "EXPORT_INFORMATIONS_DESCRIPTION": "Vous pouvez télécharger une copie de vos données",
+  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Commencez à exporter",
```
Comment (ReidyT) [inline src/langs/fr.json:112]:
```suggestion
  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Demandez vos données",
```

**Review by ReidyT — APPROVED** [review]
Thanks for this PR, it looks good to me, I just let some suggestions 😄. Also, I'm wondering if the button is disabled when the user refresh the page and the email is not sent yet.

File: package.json (hunk 1 above)
Comment (pyphilia) [reply package.json:17]: 3.22.0 is ready for you!

File: cypress/support/server.ts [hunk 8]
```diff
@@ -293,3 +294,19 @@ export const mockUpdateEmail = (shouldThrowError: boolean): void => {
...
+    {
+      method: HttpMethod.Post,
+      url: new RegExp(`${API_HOST}/${buildExportMemberDataRoute()}`),
+    },
+    ({ reply }) => {
+      if (shouldThrowError) {
+        return reply({ statusCode: StatusCodes.BAD_REQUEST });
+      }
+
+      return reply({ statusCode: StatusCodes.OK });
```
Comment (pyphilia) [inline cypress/support/server.ts:309]:
```suggestion
      return reply({ statusCode: StatusCodes.NO_CONTENT });
```

File: src/langs/fr.json [hunk 9]
```diff
@@ -106,5 +106,10 @@
   "MEMBER_STORAGE_PARENT_FOLDER": "Dossier parent",
   "NO_PARENT": "-",
   "STORAGE_FILES_ERROR": "Oops, une erreur est survenue lors de la récupération de la liste des fichiers. Veuillez réessayer plus tard.",
-  "STORAGE_FILES_EMPTY": "Vous n'avez pas téléchargé de fichier."
+  "STORAGE_FILES_EMPTY": "Vous n'avez pas téléchargé de fichier.",
+  "EXPORT_INFORMATIONS_TITLE": "Export des données",
```
Comment (pyphilia) [inline src/langs/fr.json:110]:
```suggestion
  "EXPORT_INFORMATIONS_TITLE": "Télécharger une copie de vos données",
```

File: src/langs/fr.json (hunk 7 above)
Comment (pyphilia) [inline src/langs/fr.json:112]:
```suggestion
  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Télécharger vos données",
```

File: src/langs/fr.json (hunk 5 above)
Comment (pyphilia) [inline src/langs/fr.json:113]:
```suggestion
  "EXPORT_SUCCESS_MESSAGE": "Votre demande a été envoyée. Vous recevrez un email avec vos données dans quelques minutes.",
```

File: src/langs/en.json (hunk 6 above)
Comment (pyphilia) [reply src/langs/en.json:112]:
```suggestion
  "EXPORT_INFORMATIONS_BUTTON_TEXT": "Download your data",
```

File: src/langs/en.json [hunk 10]
```diff
@@ -106,5 +106,10 @@
   "MEMBER_STORAGE_PARENT_FOLDER": "Parent folder",
   "NO_PARENT": "-",
   "STORAGE_FILES_ERROR": "Oops, an error occurred while fetching storage files. Please try again later.",
-  "STORAGE_FILES_EMPTY": "You didn't upload any file."
+  "STORAGE_FILES_EMPTY": "You didn't upload any file.",
+  "EXPORT_INFORMATIONS_TITLE": "Export data",
```
Comment (pyphilia) [inline src/langs/en.json:110]:
```suggestion
  "EXPORT_INFORMATIONS_TITLE": "Download a copy of your data",
```

File: src/langs/fr.json [hunk 11]
```diff
@@ -106,5 +106,10 @@
...
+  "EXPORT_INFORMATIONS_DESCRIPTION": "Vous pouvez télécharger une copie de vos données",
```
Comment (pyphilia) [inline src/langs/fr.json:111]:
```suggestion
  "EXPORT_INFORMATIONS_DESCRIPTION": "Vous pouvez télécharger une copie de vos données.",
```

**Review by pyphilia — CHANGES_REQUESTED** [review]
LGTM! I suggested also changes for the texts, please take the best solution between my suggestions and thibault's 🎻

**Review by pyphilia — APPROVED** [review]

**Pull request merged**